diff --git a/progs/tic.c b/progs/tic.c
index 1c0ff0b..47f7eed 100644
--- a/progs/tic.c
+++ b/progs/tic.c
@@ -389,7 +389,7 @@ open_tempfile(char *filename)
 {
     FILE *result = NULL;
 
-    _nc_STRCPY(filename, "/tmp/XXXXXX", PATH_MAX);
+    _nc_STRCPY(filename, "@TERMUX_PREFIX_CLASSICAL@/tmp/XXXXXX", PATH_MAX);
 #if HAVE_MKSTEMP
     {
 	int oldmask = (int) umask(077);
@@ -922,7 +922,7 @@ main(int argc, char *argv[])
     } else {
 	if (infodump == TRUE) {
 	    /* captoinfo's no-argument case */
-	    source_file = "/etc/termcap";
+	    source_file = "@TERMUX_PREFIX@/etc/termcap";
 	    if ((termcap = getenv("TERMCAP")) != NULL
 		&& (namelst = make_namelist(getenv("TERM"))) != NULL) {
 		if (access(termcap, F_OK) == 0) {
diff --git a/progs/tset.c b/progs/tset.c
index ed9ee28..38561d1 100644
--- a/progs/tset.c
+++ b/progs/tset.c
@@ -579,8 +579,8 @@ get_termcap_entry(int fd, const char *userarg)
 	    goto map;
 	}
 #else
-	if ((fp = fopen("/etc/ttytype", "r")) != 0
-	    || (fp = fopen("/etc/ttys", "r")) != 0) {
+	if ((fp = fopen("@TERMUX_PREFIX@/etc/ttytype", "r")) != 0
+	    || (fp = fopen("@TERMUX_PREFIX@/etc/ttys", "r")) != 0) {
 	    char buffer[BUFSIZ];
 	    char *s, *t, *d;
 
