--- src/gcc/gcc/explow.h	2024-08-01 11:17:14.000000000 +0300
+++ src/gcc/gcc/explow.h	2024-11-24 17:01:21.000000000 +0300
@@ -42,6 +42,10 @@
    Args are mode (in case value is a constant) and the value.  */
 extern rtx force_reg (machine_mode, rtx);
 
+extern rtx force_subreg (machine_mode, rtx, machine_mode, poly_uint64);
+extern rtx force_lowpart_subreg (machine_mode, rtx, machine_mode);
+extern rtx force_highpart_subreg (machine_mode, rtx, machine_mode);
+
 /* Return given rtx, copied into a new temp reg if it was in memory.  */
 extern rtx force_not_mem (rtx);
 
