diff --git a/autosetup/jimsh0.c b/autosetup/jimsh0.c
index 0f0a890..612b56d 100644
--- a/autosetup/jimsh0.c
+++ b/autosetup/jimsh0.c
@@ -23812,7 +23812,7 @@ int Jim_MakeTempFile(Jim_Interp *interp, const char *filename_template, int unli
     if (filename_template == NULL) {
         const char *tmpdir = getenv("TMPDIR");
         if (tmpdir == NULL || *tmpdir == '\0' || access(tmpdir, W_OK) != 0) {
-            tmpdir = "/tmp/";
+            tmpdir = "@TERMUX_PREFIX_CLASSICAL@/tmp/";
         }
         filenameObj = Jim_NewStringObj(interp, tmpdir, -1);
         if (tmpdir[0] && tmpdir[strlen(tmpdir) - 1] != '/') {
diff --git a/ext/wasm/libcmpp.c b/ext/wasm/libcmpp.c
index 717f56e..ae6d841 100644
--- a/ext/wasm/libcmpp.c
+++ b/ext/wasm/libcmpp.c
@@ -24,7 +24,7 @@
 #define CMPP_LIB_CONFIG_TIMESTAMP "2026-03-08 15:32 GMT"
 #define CMPP_VERSION CMPP_LIB_VERSION " " CMPP_LIB_VERSION_HASH " @ " CMPP_LIB_VERSION_TIMESTAMP
 #define CMPP_PLATFORM_EXT_DLL ".so"
-#define CMPP_MODULE_PATH ".:/usr/local/lib/cmpp"
+#define CMPP_MODULE_PATH ".:@TERMUX_PREFIX@/lib/cmpp"
 
 #if !defined(NET_WANDERINGHORSE_CMPP_H_INCLUDED_)
 #define NET_WANDERINGHORSE_CMPP_H_INCLUDED_
diff --git a/src/os_unix.c b/src/os_unix.c
index 927b8eb..e9fec55 100644
--- a/src/os_unix.c
+++ b/src/os_unix.c
@@ -6262,9 +6262,7 @@ static int fillInUnixFile(
 static const char *azTempDirs[] = {
   0,
   0,
-  "/var/tmp",
-  "/usr/tmp",
-  "/tmp",
+  "@TERMUX_PREFIX_CLASSICAL@/tmp",
   "."
 };
 
@@ -7447,7 +7445,7 @@ static int proxyGetLockPath(const char *dbPath, char *lPath, size_t maxLen){
     len = strlcat(lPath, "sqliteplocks", maxLen);  
   }
 # else
-  len = strlcpy(lPath, "/tmp/", maxLen);
+  len = strlcpy(lPath, "@TERMUX_PREFIX_CLASSICAL@/tmp/", maxLen);
 # endif
 #endif
 
diff --git a/src/os_win.c b/src/os_win.c
index ba64512..1cafccc 100644
--- a/src/os_win.c
+++ b/src/os_win.c
@@ -5107,9 +5107,7 @@ static int winGetTempname(sqlite3_vfs *pVfs, char **pzBuf){
        0, /* getenv("TMP") */
        0, /* getenv("TEMP") */
        0, /* getenv("USERPROFILE") */
-       "/var/tmp",
-       "/usr/tmp",
-       "/tmp",
+       "@TERMUX_PREFIX_CLASSICAL@/tmp",
        ".",
        0        /* List terminator */
     };
diff --git a/tool/lemon.c b/tool/lemon.c
index 9071e77..9206126 100644
--- a/tool/lemon.c
+++ b/tool/lemon.c
@@ -3666,7 +3666,7 @@ PRIVATE char *pathsearch(char *argv0, char *name, int modemask)
     *cp = c;
   }else{
     pathlist = getenv("PATH");
-    if( pathlist==0 ) pathlist = ".:/bin:/usr/bin";
+    if( pathlist==0 ) pathlist = ".:@TERMUX_PREFIX@/bin";
     pathbuf = (char *) lemon_malloc( lemonStrlen(pathlist) + 1 );
     path = (char *)lemon_malloc( lemonStrlen(pathlist)+lemonStrlen(name)+2 );
     if( (pathbuf != 0) && (path!=0) ){
