diff --git a/CMakeLists.txt b/CMakeLists.txt
index 86f9519..a24167b 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -717,14 +717,14 @@ if(NOT _x86 AND NOT _x86_64)
     RUNTIME DESTINATION bin)
   if(NOT NO_CONF_INSTALL)
     configure_file(system/box86.conf.cmake system/box86.conf)
-    install(FILES ${CMAKE_BINARY_DIR}/system/box86.conf DESTINATION /etc/binfmt.d/)
-    install(FILES ${CMAKE_SOURCE_DIR}/system/box86.box86rc DESTINATION /etc/)
+    install(FILES ${CMAKE_BINARY_DIR}/system/box86.conf DESTINATION @TERMUX_PREFIX@/etc/binfmt.d/)
+    install(FILES ${CMAKE_SOURCE_DIR}/system/box86.box86rc DESTINATION @TERMUX_PREFIX@/etc/)
   endif()
   if(NOT NO_LIB_INSTALL)
-    install(FILES ${CMAKE_SOURCE_DIR}/x86lib/libstdc++.so.6 DESTINATION /usr/lib/i386-linux-gnu/)
-    install(FILES ${CMAKE_SOURCE_DIR}/x86lib/libstdc++.so.5 DESTINATION /usr/lib/i386-linux-gnu/)
-    install(FILES ${CMAKE_SOURCE_DIR}/x86lib/libgcc_s.so.1 DESTINATION /usr/lib/i386-linux-gnu/)
-    install(FILES ${CMAKE_SOURCE_DIR}/x86lib/libpng12.so.0 DESTINATION /usr/lib/i386-linux-gnu/)
+    install(FILES ${CMAKE_SOURCE_DIR}/x86lib/libstdc++.so.6 DESTINATION @TERMUX_PREFIX@/lib/i386-linux-gnu/)
+    install(FILES ${CMAKE_SOURCE_DIR}/x86lib/libstdc++.so.5 DESTINATION @TERMUX_PREFIX@/lib/i386-linux-gnu/)
+    install(FILES ${CMAKE_SOURCE_DIR}/x86lib/libgcc_s.so.1 DESTINATION @TERMUX_PREFIX@/lib/i386-linux-gnu/)
+    install(FILES ${CMAKE_SOURCE_DIR}/x86lib/libpng12.so.0 DESTINATION @TERMUX_PREFIX@/lib/i386-linux-gnu/)
   endif()
 endif()
 
diff --git a/src/main.c b/src/main.c
index 0d7e7f2..69cc760 100755
--- a/src/main.c
+++ b/src/main.c
@@ -880,20 +880,14 @@ void LoadEnvVars(box86context_t *context)
         AddPath("/mnt/utmp/box86/lib/i386-linux-gnu", &context->box86_ld_lib, 1);
     //TODO: add relative path to box86 location
 #endif
-    if(FileExist("/lib/box86", 0))
-        AddPath("/lib/box86", &context->box86_ld_lib, 1);
-    if(FileExist("/usr/lib/box86", 0))
-        AddPath("/usr/lib/box86", &context->box86_ld_lib, 1);
-    if(FileExist("/lib/i386-linux-gnu", 0))
-        AddPath("/lib/i386-linux-gnu", &context->box86_ld_lib, 1);
-    if(FileExist("/usr/lib/i386-linux-gnu", 0))
-        AddPath("/usr/lib/i386-linux-gnu", &context->box86_ld_lib, 1);
-    if(FileExist("/lib/i686-pc-linux-gnu", 0))
-        AddPath("/lib/i686-pc-linux-gnu", &context->box86_ld_lib, 1);
-    if(FileExist("/usr/lib/i686-pc-linux-gnu", 0))
-        AddPath("/usr/lib/i686-pc-linux-gnu", &context->box86_ld_lib, 1);
-    if(FileExist("/usr/lib32", 0))
-        AddPath("/usr/lib32", &context->box86_ld_lib, 1);
+    if(FileExist("@TERMUX_PREFIX@/lib/box86", 0))
+        AddPath("@TERMUX_PREFIX@/lib/box86", &context->box86_ld_lib, 1);
+    if(FileExist("@TERMUX_PREFIX@/lib/i386-linux-gnu", 0))
+        AddPath("@TERMUX_PREFIX@/lib/i386-linux-gnu", &context->box86_ld_lib, 1);
+    if(FileExist("@TERMUX_PREFIX@/lib/i686-pc-linux-gnu", 0))
+        AddPath("@TERMUX_PREFIX@/lib/i686-pc-linux-gnu", &context->box86_ld_lib, 1);
+    if(FileExist("@TERMUX_PREFIX@/lib32", 0))
+        AddPath("@TERMUX_PREFIX@/lib32", &context->box86_ld_lib, 1);
     if(getenv("LD_LIBRARY_PATH"))
         PrependList(&context->box86_ld_lib, getenv("LD_LIBRARY_PATH"), 1);   // in case some of the path are for x86 world
     if(getenv("BOX86_EMULATED_LIBS")) {
@@ -1101,8 +1095,8 @@ void endBox86()
 
 static void load_rcfiles()
 {
-    if(FileExist("/etc/box86.box86rc", IS_FILE))
-        LoadRCFile("/etc/box86.box86rc");
+    if(FileExist("@TERMUX_PREFIX@/etc/box86.box86rc", IS_FILE))
+        LoadRCFile("@TERMUX_PREFIX@/etc/box86.box86rc");
     #ifdef PANDORA
     if(FileExist("/mnt/utmp/codeblocks/usr/etc/box86.box86rc", IS_FILE))
         LoadRCFile("/mnt/utmp/codeblocks/usr/etc/box86.box86rc");
@@ -1174,7 +1168,7 @@ int main(int argc, const char **argv, char **env)
         PrintHelp();
         return 1;
     }
-    if(argc>1 && !strcmp(argv[1], "/usr/bin/gdb"))
+    if(argc>1 && !strcmp(argv[1], "@TERMUX_PREFIX@/bin/gdb"))
         exit(0);
     // uname -m is redirected to box86 -m
     if(argc==2 && (!strcmp(argv[1], "-m") || !strcmp(argv[1], "-p") || !strcmp(argv[1], "-i")))
diff --git a/src/tools/fileutils.c b/src/tools/fileutils.c
index ee7a1a2..46ca89e 100755
--- a/src/tools/fileutils.c
+++ b/src/tools/fileutils.c
@@ -17,9 +17,9 @@
 
 static const char* x86lib  = "\x7f" "ELF" "\x01" "\x01" "\x01" "\x03" "\x00" "\x00" "\x00" "\x00" "\x00" "\x00" "\x00" "\x00" "\x02" "\x00" "\x03" "\x00";
 static const char* x64lib  = "\x7f" "ELF" "\x02" "\x01" "\x01" "\x03" "\x00" "\x00" "\x00" "\x00" "\x00" "\x00" "\x00" "\x00" "\x02" "\x00" "\x3e" "\x00";
-static const char* bashsign= "#!/bin/bash";
-static const char* shsign  = "#!/bin/sh";
-static const char* bashsign2="#!/usr/bin/env bash";
+static const char* bashsign= "#!@TERMUX_PREFIX@/bin/bash";
+static const char* shsign  = "#!@TERMUX_PREFIX@/bin/sh";
+static const char* bashsign2="#!@TERMUX_PREFIX@/bin/env bash";
 
 int FileExist(const char* filename, int flags)
 {
@@ -126,11 +126,8 @@ const char* GetTmpDir() {
     if ((tmpdir = getenv ("TMPDIR")) != NULL) return tmpdir;
     if ((tmpdir = getenv ("TEMP")) != NULL)   return tmpdir;
     if ((tmpdir = getenv ("TMP")) != NULL)    return tmpdir;
-    if(FileExist("/tmp", 0))                  return "/tmp";
-    if(FileExist("/var/tmp", 0))              return "/var/tmp";
-    if(FileExist("/usr/tmp", 0))              return "/usr/tmp";
 
-    return "/tmp";  // meh...
+    return "@TERMUX_PREFIX_CLASSICAL@/tmp";
 }
 
 char* LowerCase(const char* s) {
diff --git a/src/wrapped/wrappedd3dadapter9.c b/src/wrapped/wrappedd3dadapter9.c
index 7b13da1..0ed698f 100644
--- a/src/wrapped/wrappedd3dadapter9.c
+++ b/src/wrapped/wrappedd3dadapter9.c
@@ -98,7 +98,7 @@ static int my_GetDirect3D(x86emu_t* emu, void* This, void*** ppD3D9);
 
 const char* d3dadapter9Name = "d3dadapter9.so.1";
 #define LIBNAME d3dadapter9
-#define ALTNAME "/usr/lib/arm-linux-gnueabihf/d3d/d3dadapter9.so.1"
+#define ALTNAME "@TERMUX_PREFIX@/lib/arm-linux-gnueabihf/d3d/d3dadapter9.so.1"
 
 typedef void* (*pFp_t)(void*);
 
diff --git a/src/wrapped/wrappedlibc.c b/src/wrapped/wrappedlibc.c
index 6c0c64a..972096f 100755
--- a/src/wrapped/wrappedlibc.c
+++ b/src/wrapped/wrappedlibc.c
@@ -2526,7 +2526,7 @@ EXPORT void* my_getpwuid(x86emu_t* emu, uint32_t uid)
         fakepwd.pw_gid = 0;
         fakepwd.pw_gecos = strdup("root");
         fakepwd.pw_dir = getenv("HOME");
-        fakepwd.pw_shell = strdup("/bin/bash");
+        fakepwd.pw_shell = strdup("@TERMUX_PREFIX@/bin/bash");
     }
 
     return ret ? ret : (void*)&fakepwd;
diff --git a/system/box86.conf.cmake b/system/box86.conf.cmake
index f61fc16..9708da8 100755
--- a/system/box86.conf.cmake
+++ b/system/box86.conf.cmake
@@ -1,2 +1,2 @@
-# /etc/binfmt.d/box86.conf
+# @TERMUX_PREFIX@/etc/binfmt.d/box86.conf
 :x86:M::\x7fELF\x01\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\x03\x00:\xff\xff\xff\xff\xff\xff\xff\x00\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\xff:${CMAKE_INSTALL_PREFIX}/bin/${BOX86}:
