From https://github.com/xMeM/termux-packages/commit/57b1bb44c9eed341c700105efed93f9fd8bc34a6
diff --git a/src/gallium/drivers/freedreno/freedreno_screen.c b/src/gallium/drivers/freedreno/freedreno_screen.c
index 1c6401aa968..cf45d1def86 100644
--- a/src/gallium/drivers/freedreno/freedreno_screen.c
+++ b/src/gallium/drivers/freedreno/freedreno_screen.c
@@ -587,8 +587,7 @@ fd_screen_get_param(struct pipe_screen *pscreen, enum pipe_cap param)
    case PIPE_CAP_QUERY_TIMESTAMP:
    case PIPE_CAP_QUERY_TIME_ELAPSED:
       /* only a4xx, requires new enough kernel so we know max_freq: */
-      return (screen->max_freq > 0) &&
-             (is_a4xx(screen) || is_a5xx(screen) || is_a6xx(screen));
+      return (is_a4xx(screen) || is_a5xx(screen) || is_a6xx(screen));
    case PIPE_CAP_TIMER_RESOLUTION:
       return ticks_to_ns(1);
    case PIPE_CAP_QUERY_BUFFER_OBJECT:
