--- glibc-2.39/sysdeps/nptl/_Fork.c	2021-08-02 01:33:43.000000000 +0000
+++ glibc-2.39/sysdeps/nptl/_Fork.c.path	2022-01-20 08:39:46.858400434 +0000
@@ -44,8 +44,6 @@
       self->robust_prev = &self->robust_head;
 #endif
       self->robust_head.list = &self->robust_head;
-      INTERNAL_SYSCALL_CALL (set_robust_list, &self->robust_head,
-			     sizeof (struct robust_list_head));
     }
   return pid;
 }
