--- glibc-2.38/sysdeps/nptl/dl-tls_init_tp.c	2021-08-02 01:33:43.000000000 +0000
+++ glibc-2.38/sysdeps/nptl/dl-tls_init_tp.c.path	2022-01-20 08:42:04.148400381 +0000
@@ -80,14 +80,9 @@
     pd->robust_head.futex_offset = (offsetof (pthread_mutex_t, __data.__lock)
                                     - offsetof (pthread_mutex_t,
                                                 __data.__list.__next));
-    int res = INTERNAL_SYSCALL_CALL (set_robust_list, &pd->robust_head,
-                                     sizeof (struct robust_list_head));
-    if (!INTERNAL_SYSCALL_ERROR_P (res))
-      {
 #ifndef __ASSUME_SET_ROBUST_LIST
         __nptl_set_robust_list_avail = true;
 #endif
-      }
   }
 
   /* Set initial thread's stack block from 0 up to __libc_stack_end.
