--- glibc-2.39/nptl/pthread_create.c	2021-08-02 01:33:43.000000000 +0000
+++ glibc-2.39/nptl/pthread_create.c.path	2022-01-20 08:16:09.293375256 +0000
@@ -366,16 +366,6 @@
   /* Initialize pointers to locale data.  */
   __ctype_init ();
 
-#ifndef __ASSUME_SET_ROBUST_LIST
-  if (__nptl_set_robust_list_avail)
-#endif
-    {
-      /* This call should never fail because the initial call in init.c
-	 succeeded.  */
-      INTERNAL_SYSCALL_CALL (set_robust_list, &pd->robust_head,
-			     sizeof (struct robust_list_head));
-    }
-
   /* This is where the try/finally block should be created.  For
      compilers without that support we do use setjmp.  */
   struct pthread_unwind_buf unwind_buf;
