diff --git a/gnulib-tests/test-read-file.c b/gnulib-tests/test-read-file.c
index 98dd681..9454edc 100644
--- a/gnulib-tests/test-read-file.c
+++ b/gnulib-tests/test-read-file.c
@@ -25,7 +25,7 @@
 
 #include "macros.h"
 
-#define FILE1 "/etc/resolv.conf"
+#define FILE1 "@TERMUX_PREFIX@/etc/resolv.conf"
 #define FILE2 "/dev/null"
 
 static int
diff --git a/lib/mountlist.c b/lib/mountlist.c
index 77ea8ce..69e3a95 100644
--- a/lib/mountlist.c
+++ b/lib/mountlist.c
@@ -780,7 +780,7 @@ read_file_system_list (bool need_fs_type)
 
 #if defined MOUNTED_FREAD_FSTYP /* (obsolete) SVR3 */
   {
-    char *table = "/etc/mnttab";
+    char *table = "@TERMUX_PREFIX@/etc/mnttab";
 
     FILE *fp = fopen (table, "re");
     if (fp == NULL)
@@ -887,7 +887,7 @@ read_file_system_list (bool need_fs_type)
        for this file name, we should use their macro name instead.
        (Why not just lock MNTTAB directly?  We don't know.)  */
 #  ifndef MNTTAB_LOCK
-#   define MNTTAB_LOCK "/etc/.mnttab.lock"
+#   define MNTTAB_LOCK "@TERMUX_PREFIX@/etc/.mnttab.lock"
 #  endif
     lockfd = open (MNTTAB_LOCK, O_RDONLY | O_CLOEXEC);
     if (0 <= lockfd)
diff --git a/lib/tmpdir.c b/lib/tmpdir.c
index a1473e2..81a2a49 100644
--- a/lib/tmpdir.c
+++ b/lib/tmpdir.c
@@ -35,7 +35,7 @@
 # ifdef _P_tmpdir /* native Windows */
 #  define P_tmpdir _P_tmpdir
 # else
-#  define P_tmpdir "/tmp"
+#  define P_tmpdir "@TERMUX_PREFIX_CLASSICAL@/tmp"
 # endif
 #endif
 
@@ -131,8 +131,8 @@ path_search (char *tmpl, size_t tmpl_len, const char *dir, const char *pfx,
 #endif
       if (direxists (P_tmpdir))
         dir = P_tmpdir;
-      else if (strcmp (P_tmpdir, "/tmp") != 0 && direxists ("/tmp"))
-        dir = "/tmp";
+      else if (strcmp (P_tmpdir, "@TERMUX_PREFIX_CLASSICAL@/tmp") != 0 && direxists ("@TERMUX_PREFIX_CLASSICAL@/tmp"))
+        dir = "@TERMUX_PREFIX_CLASSICAL@/tmp";
       else
         {
           __set_errno (ENOENT);
diff --git a/src/basename.c b/src/basename.c
index 96bbb71..6fc0153 100644
--- a/src/basename.c
+++ b/src/basename.c
@@ -73,7 +73,7 @@ If specified, also remove a trailing SUFFIX.\n\
       printf (_("\
 \n\
 Examples:\n\
-  %s /usr/bin/sort          -> \"sort\"\n\
+  %s @TERMUX_PREFIX@/bin/sort          -> \"sort\"\n\
   %s include/stdio.h .h     -> \"stdio\"\n\
   %s -s .h include/stdio.h  -> \"stdio\"\n\
   %s -a any/str1 any/str2   -> \"str1\" followed by \"str2\"\n\
diff --git a/src/chroot.c b/src/chroot.c
index 9a52625..ee26fb3 100644
--- a/src/chroot.c
+++ b/src/chroot.c
@@ -207,7 +207,7 @@ Run COMMAND with root directory set to NEWROOT.\n\
       oputs (VERSION_OPTION_DESCRIPTION);
       fputs (_("\
 \n\
-If no command is given, run '\"$SHELL\" -i' (default: '/bin/sh -i').\n\
+If no command is given, run '\"$SHELL\" -i' (default: '@TERMUX_PREFIX@/bin/sh -i').\n\
 "), stdout);
       emit_exec_status (PROGRAM_NAME);
       emit_ancillary_info (PROGRAM_NAME);
@@ -339,7 +339,7 @@ main (int argc, char **argv)
       /* No command.  Run an interactive shell.  */
       char *shell = getenv ("SHELL");
       if (shell == NULL)
-        shell = bad_cast ("/bin/sh");
+        shell = bad_cast ("@TERMUX_PREFIX@/bin/sh");
       argv[0] = shell;
       argv[1] = bad_cast ("-i");
       argv[2] = NULL;
diff --git a/src/dirname.c b/src/dirname.c
index dbd21d6..e0b9d08 100644
--- a/src/dirname.c
+++ b/src/dirname.c
@@ -64,7 +64,7 @@ removed; if NAME contains no /'s, output '.' (meaning the current directory).\n\
       printf (_("\
 \n\
 Examples:\n\
-  %s /usr/bin/          -> \"/usr\"\n\
+  %s @TERMUX_PREFIX@/bin/          -> \"@TERMUX_PREFIX@\"\n\
   %s dir1/str dir2/str  -> \"dir1\" followed by \"dir2\"\n\
   %s stdio.h            -> \".\"\n\
 "),
diff --git a/src/mktemp.c b/src/mktemp.c
index 1fa7972..db65889 100644
--- a/src/mktemp.c
+++ b/src/mktemp.c
@@ -92,7 +92,7 @@ Files are created u+rw, and directories u+rwx, minus umask restrictions.\n\
       oputs (_("\
   -p DIR, --tmpdir[=DIR]\n\
          interpret TEMPLATE relative to DIR;\n\
-         if DIR is not specified, use $TMPDIR if set, else /tmp.\n\
+         if DIR is not specified, use $TMPDIR if set, else @TERMUX_PREFIX_CLASSICAL@/tmp.\n\
          With this option, TEMPLATE must not be an absolute name;\n\
          unlike with -t, TEMPLATE may contain slashes,\n\
          but mktemp creates only the final component\n\
@@ -101,7 +101,7 @@ Files are created u+rw, and directories u+rwx, minus umask restrictions.\n\
   -t\n\
          interpret TEMPLATE as a single file name component,\n\
          relative to a directory: $TMPDIR, if set;\n\
-         else the directory specified via -p; else /tmp [deprecated]\n\
+         else the directory specified via -p; else @TERMUX_PREFIX_CLASSICAL@/tmp [deprecated]\n\
 "));
       oputs (HELP_OPTION_DESCRIPTION);
       oputs (VERSION_OPTION_DESCRIPTION);
@@ -283,7 +283,7 @@ main (int argc, char **argv)
           else if (dest_dir_arg && *dest_dir_arg)
             dest_dir = dest_dir_arg;
           else
-            dest_dir = "/tmp";
+            dest_dir = "@TERMUX_PREFIX_CLASSICAL@/tmp";
 
           if (last_component (template) != template)
             error (EXIT_FAILURE, 0,
@@ -297,7 +297,7 @@ main (int argc, char **argv)
           else
             {
               char *env = getenv ("TMPDIR");
-              dest_dir = (env && *env ? env : "/tmp");
+              dest_dir = (env && *env ? env : "@TERMUX_PREFIX_CLASSICAL@/tmp");
             }
           if (IS_ABSOLUTE_FILE_NAME (template))
             error (EXIT_FAILURE, 0,
diff --git a/src/sort.c b/src/sort.c
index 510c674..ed880ab 100644
--- a/src/sort.c
+++ b/src/sort.c
@@ -83,7 +83,7 @@ struct rlimit { size_t rlim_cur; };
 #define UCHAR_LIM (UCHAR_MAX + 1)
 
 #ifndef DEFAULT_TMPDIR
-# define DEFAULT_TMPDIR "/tmp"
+# define DEFAULT_TMPDIR "@TERMUX_PREFIX_CLASSICAL@/tmp"
 #endif
 
 /* Maximum number of lines to merge every time a NODE is taken from
diff --git a/src/split.c b/src/split.c
index a8e6f8b..34e8467 100644
--- a/src/split.c
+++ b/src/split.c
@@ -594,7 +594,7 @@ create (char const *name)
 
       char const *shell_prog = getenv ("SHELL");
       if (shell_prog == NULL)
-        shell_prog = "/bin/sh";
+        shell_prog = "@TERMUX_PREFIX@/bin/sh";
       char const *const argv[] = { last_component (shell_prog), "-c",
                                    filter_command, NULL };
 
