diff --git a/src/main.c b/src/main.c
index db33d9a..1870884 100644
--- a/src/main.c
+++ b/src/main.c
@@ -1161,12 +1161,10 @@ void LoadEnvVars(box64context_t *context)
     // check BOX64_LD_LIBRARY_PATH and load it
     LoadEnvPath(&context->box64_ld_lib, ".:lib:lib64:x86_64:bin64:libs64", "BOX64_LD_LIBRARY_PATH");
     #ifndef TERMUX
-    if(FileExist("/lib/x86_64-linux-gnu", 0))
-        AddPath("/lib/x86_64-linux-gnu", &context->box64_ld_lib, 1);
-    if(FileExist("/usr/lib/x86_64-linux-gnu", 0))
-        AddPath("/usr/lib/x86_64-linux-gnu", &context->box64_ld_lib, 1);
-    if(FileExist("/usr/x86_64-linux-gnu/lib", 0))
-        AddPath("/usr/x86_64-linux-gnu/lib", &context->box64_ld_lib, 1);
+    if(FileExist("@TERMUX_PREFIX@/lib/x86_64-linux-gnu", 0))
+        AddPath("@TERMUX_PREFIX@/lib/x86_64-linux-gnu", &context->box64_ld_lib, 1);
+    if(FileExist("@TERMUX_PREFIX@/x86_64-linux-gnu/lib", 0))
+        AddPath("@TERMUX_PREFIX@/x86_64-linux-gnu/lib", &context->box64_ld_lib, 1);
     #else
     //TODO: Add Termux Library Path - Lily
     if(FileExist("/data/data/com.termux/files/usr/lib/x86_64-linux-gnu", 0))
@@ -1426,8 +1424,8 @@ static void free_contextargv()
 static void load_rcfiles()
 {
     #ifndef TERMUX
-    if(FileExist("/etc/box64.box64rc", IS_FILE))
-        LoadRCFile("/etc/box64.box64rc");
+    if(FileExist("@TERMUX_PREFIX@/etc/box64.box64rc", IS_FILE))
+        LoadRCFile("@TERMUX_PREFIX@/etc/box64.box64rc");
     #else
     if(FileExist("/data/data/com.termux/files/usr/etc/box64.box64rc", IS_FILE))
         LoadRCFile("/data/data/com.termux/files/usr/etc/box64.box64rc");
@@ -1459,7 +1457,7 @@ int main(int argc, const char **argv, char **env) {
         printf("See 'box64 --help' for more information.\n");
         exit(0);
     }
-    if(argc>1 && !strcmp(argv[1], "/usr/bin/gdb") && getenv("BOX64_TRACE_FILE"))
+    if(argc>1 && !strcmp(argv[1], "@TERMUX_PREFIX@/bin/gdb") && getenv("BOX64_TRACE_FILE"))
         exit(0);
     // uname -m is redirected to box64 -m
     if(argc==2 && (!strcmp(argv[1], "-m") || !strcmp(argv[1], "-p") || !strcmp(argv[1], "-i")))
diff --git a/src/steam.c b/src/steam.c
index b221b54..d0af6e4 100644
--- a/src/steam.c
+++ b/src/steam.c
@@ -41,13 +41,13 @@ void pressure_vessel(int argc, const char** argv, int nextarg, const char* prog)
                 if(is_usr) {
                     // transform RESSURE_VESSEL_APP_LD_LIBRARY_PATH to BOX86_ / BOX64_ LD_LIBRARY_PATH
                     char tmp[strlen(argv[nextarg])+150];
-                    strcpy(tmp, "BOX86_LD_LIBRARY_PATH=/lib/box86:/usr/lib/box86:/lib/i386-linux-gnu:/usr/lib/i386-linux-gnu:/usr/lib/i686-pc-linux-gnu:/usr/lib32:");
+                    strcpy(tmp, "BOX86_LD_LIBRARY_PATH=@TERMUX_PREFIX@/lib/box86:@TERMUX_PREFIX@/lib/i386-linux-gnu:@TERMUX_PREFIX@/lib/i686-pc-linux-gnu:@TERMUX_PREFIX@/lib32:");
                     strcat(tmp, argv[nextarg]+strlen("--env-if-host=STEAM_RUNTIME_LIBRARY_PATH="));
                     char *p = strchr(tmp, '=');
                     *p ='\0'; ++p;
                     setenv(tmp, p, 1);
                     printf_log(LOG_DEBUG, "setenv(%s, %s, 1)\n", tmp, p);
-                    strcpy(tmp, "BOX64_LD_LIBRARY_PATH=/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu:");
+                    strcpy(tmp, "BOX64_LD_LIBRARY_PATH=@TERMUX_PREFIX@/lib/x86_64-linux-gnu:");
                     strcat(tmp, argv[nextarg]+strlen("--env-if-host=STEAM_RUNTIME_LIBRARY_PATH="));
                     p = strchr(tmp, '=');
                     *p ='\0'; ++p;
@@ -173,4 +173,4 @@ void pressure_vessel(int argc, const char** argv, int nextarg, const char* prog)
         //waitpid(v, &wstatus, 0);
         exit(0);
     }
-}
\ No newline at end of file
+}
diff --git a/src/tools/fileutils.c b/src/tools/fileutils.c
index d6d1556..f272a9c 100644
--- a/src/tools/fileutils.c
+++ b/src/tools/fileutils.c
@@ -127,11 +127,8 @@ const char* GetTmpDir() {
     if ((tmpdir = getenv ("TMPDIR")) != NULL) return tmpdir;
     if ((tmpdir = getenv ("TEMP")) != NULL)   return tmpdir;
     if ((tmpdir = getenv ("TMP")) != NULL)    return tmpdir;
-    if(FileExist("/tmp", 0))                  return "/tmp";
-    if(FileExist("/var/tmp", 0))              return "/var/tmp";
-    if(FileExist("/usr/tmp", 0))              return "/usr/tmp";
 
-    return "/tmp";  // meh...
+    return "@TERMUX_PREFIX_CLASSICAL@/tmp";  // meh...
 }
 
 char* LowerCase(const char* s) {
diff --git a/src/wrapped/wrappedd3dadapter9.c b/src/wrapped/wrappedd3dadapter9.c
index 3701bba..50161fb 100644
--- a/src/wrapped/wrappedd3dadapter9.c
+++ b/src/wrapped/wrappedd3dadapter9.c
@@ -98,7 +98,7 @@ static int my_GetDirect3D(x64emu_t* emu, void* This, void*** ppD3D9);
 
 const char* d3dadapter9Name = "d3dadapter9.so.1";
 #define LIBNAME d3dadapter9
-#define ALTNAME "/usr/lib/aarch64-linux-gnu/d3d/d3dadapter9.so.1"
+#define ALTNAME "@TERMUX_PREFIX@/lib/aarch64-linux-gnu/d3d/d3dadapter9.so.1"
 
 typedef void* (*pFp_t)(void*);
 
diff --git a/src/wrapped/wrappedlibdl.c b/src/wrapped/wrappedlibdl.c
index 217ae01..d83d1f5 100644
--- a/src/wrapped/wrappedlibdl.c
+++ b/src/wrapped/wrappedlibdl.c
@@ -99,7 +99,7 @@ void* my_dlopen(x64emu_t* emu, void *filename, int flag)
             if(sys)
                 return sys;
         }
-        if(!strcmp(rfilename, "/usr/lib/x86_64-linux-gnu/d3d")) {
+        if(!strcmp(rfilename, "@TERMUX_PREFIX@/lib/x86_64-linux-gnu/d3d")) {
             void* sys = my_dlopen(emu, "d3dadapter9.so.1", flag);
             if(sys)
                 return sys;
